### DISCUSSION TITLE
Walk_fast output bout duration

### DISCUSSION THREAD

**Discussion by Suesmith12** [Ergo-Tools/ActiPASS#63] [Help and Q&A (processing, quality checking or statistics)] [answered]
Hi Pasan,

We have been getting quite different results for the walk_fast variable when comparing it to output from other softwares (ProcessingPAL) using a cadence of >=100. Is this difference potentially due to there being no minimum bout duration for walking fast in ActiPASS, and if so, is there a way to set a minimum bout duration please?

Thank you

**Comment by pasan-h** [comment, marked as answer]
Hi Sue,
Yes, your observation is a result of comparably very low epoch length (high time resolution) of the activity detection algorithm. In ActiPASS the effective epoch-length for activities varies between around 2 seconds up to 20 seconds depending on the activity/behaviour. For activities such as walking, ActiPASS has it's best time resolution (the lowest epoch length). For more information about this phenomenon see [Epoch length and the physical activity bout analysis: An accelerometery research issue](https://rdcu.be/dramu)

If you see [table 1 of our EHJ paper](https://doi.org/10.1093/eurheartj/ehad717), you can see for pooled ProPASS cohorts MVPA time per day is around 1.3 hrs (78 minutes). This result is exactly what you have also observed. This means we need to either reduce the underlying time resolution to be comparable to other tools, or we need to accept and address the high time resolution when we interpret results. 

Reducing the time resolution (increasing the epoch length) is relatively simple for count based algorithms. Because  It's just averaging the counts for desired epoch length. But for algorithms which output multiple behaviours, this process needs to be done differently. I think the process boils down to finding the most common activity within the increased epoch length. This process is something which we ActiPASS developers have discussed but have not yet implemented. However, ActiPASS exposes it's high resolution activity/behaviour output in CSV format if desired. If you want to do this reduction process yourselves we can discuss more.

Another approach would be to consider ActiPASS bout variables. If you consider these variables, then the best option is to use intensity-class variables such as INT34 (which is analogous to MVPA). Anyway if you are going to use ActiPASS bout variables, please read [ActiPASS Variable definitions](https://view.officeapps.live.com/op/view.aspx?src=https://github.com/Ergo-Tools/ActiPASS/raw/main/docs/Extra/ActiPASS_Variable-definitions_NewFormat_ActiPASS_Version_1.50.xlsx). 

For an example the variable **INT34_2min_bouts_TH** will give the times of MVPA bouts which are longer than 2 minutes allowing bouts break up to 20 seconds.

**Reply by pasan-h** [reply]
For more information about ActiPASS bout variables, see:
 [How physical activity (PA), sedentary behaviour (SB) or intensity-class (SED, LPA, MVPA, VPA etc.) bouts are derived by ActiPASS](https://github.com/Ergo-Tools/ActiPASS/wiki/How-bout-variables-are-derived-by-ActiPASS#how-physical-activity-pa-sedentary-behaviour-sb-or-intensity-class-sed-lpa-mvpa-vpa-etc-bouts-are-derived-by-actipass).

**Reply by Suesmith12** [reply]
Hi Pasan,
thank you so much for your quick response and your advice. The wiki page was so useful to understand this!

Our primary outcome is brisk walking (in isolation), so I think it may be best to use INT3 for this and to look at 1 minute and 2 minute bouts. 

However, if we were to take a compositional/24-h approach and use ActiPASS INT variables for this - I believe we couldn't use the bouts variables as this would not make up the necessary composition? Or can they still be used and where anything outside the included bouts is categorised/counted back into another behaviour, if that makes sense?

In the linked EHJ paper, there was no minimum bout, with acknowledgement that it may capture very short bursts of movement - was no minimum bout duration set because of the above reason? i.e., the bout variable couldn't be included in the analysis

Thanks!

**Comment by pasan-h** [comment]
> Our primary outcome is brisk walking (in isolation), so I think it may be best to use INT3 for this and to look at 1 minute and 2 minute bouts.

Very difficult question for me 😊. Regarding which bout length to use and how long breaks are allowed within bouts, I think there is no definite answer. Perhaps answer depends on physiological factors (such as time it takes for body reach steady state MET), our perception of a bout for particular behaviour/activity, compliance with existing studies/standards etc. For an example, for bouts of sitting+lying or upright behaviours, one can argue that no bout-breaks should be allowed.

> However, if we were to take a compositional/24-h approach and use ActiPASS INT variables for this - I believe we couldn't use the bouts variables as this would not make up the necessary composition? Or can they still be used and where anything outside the included bouts is categorised/counted back into another behaviour, if that makes sense?

Yes, you are correct. we cannot sum up the times of bout variables to 24 hours.  This means when we use bout variables for intensity classes, we cannot do CODA (at least that is how I understand the situation, I could be wrong though 😊)

> In the linked EHJ paper, there was no minimum bout, with acknowledgement that it may capture very short bursts of movement - was no minimum bout duration set because of the above reason? i.e., the bout variable couldn't be included in the analysis

Yes, I think so too. In other words, one design goal of Acti4/ActiPASS was to capture short bursts of activities which are hidden otherwise. Therefore a very low epoch-length is chosen intentionally. Because of this design choice we can now detect such short bursts of activities. However this also means total MVPA (INT34) time reported by ActiPASS would be considerably higher than tools with longer epoch-length. This also means there will now be many intermittent breaks within a perceived bout of given activity/behaviour.

Regarding bouts (derivation of bouts), perhaps it's worth reading this paper as well.
https://bmcresnotes.biomedcentral.com/articles/10.1186/1756-0500-7-284

Also this discussion might be of interest to you:
https://github.com/Ergo-Tools/ActiPASS/discussions/62

**Reply by Suesmith12** [reply]
Hi Pasan, thank you very much for all of that information and for the insight about using ActiPASS output in CODA - that is really informative and much appreciated! 😊

**Comment by Cfmbiemans** [comment]
Dear @pasan-h ,

I've read all comments above, but have an additionial question related to this.

I would like to clarify exactly how SitLie_T30min is defined.

**Is SitLie_T30min the total duration of continuous sitting/lying bouts that last longer than 30 minutes, including the first 30 minutes of those bouts?** Or does it include only the time beyond the initial 30 minutes?

To make my question concrete:
Suppose a person sits continuously for 90 minutes.
Would this count as SitLie_T30min = 90 minutes, because the entire sitting bout is longer than 30 minutes?
Or would it count as SitLie_T30min = 60 minutes, because the first 30 minutes are not yet considered “prolonged” sitting/lying, and only the time after 30 minutes is included?

In other words, when a continuous Sit/Lie segment exceeds 30 minutes, is the variable calculated as:

the full duration of that segment, or
only the portion exceeding 30 minutes?

I hope I made myself clear.

Thank you very much in advance.

Camille

**Reply by pasan-h** [reply]
> Dear @pasan-h ,
> 
> I've read all comments above, but have an additionial question related to this.
> 
> I would like to clarify exactly how SitLie_T30min is defined.
> 
> **Is SitLie_T30min the total duration of continuous sitting/lying bouts that last longer than 30 minutes, including the first 30 minutes of those bouts?** Or does it include only the time beyond the initial 30 minutes?
> 
> To make my question concrete: Suppose a person sits continuously for 90 minutes. Would this count as SitLie_T30min = 90 minutes, because the entire sitting bout is longer than 30 minutes? Or would it count as SitLie_T30min = 60 minutes, because the first 30 minutes are not yet considered “prolonged” sitting/lying, and only the time after 30 minutes is included?
> 
> In other words, when a continuous Sit/Lie segment exceeds 30 minutes, is the variable calculated as:
> 
> the full duration of that segment, or only the portion exceeding 30 minutes?
> 
> I hope I made myself clear.
> 
> Thank you very much in advance.
> 
> Camille

I assume you meant **SitLie_30min_bouts_TH**.  If a person sit continuously for 90 minutes, and no sitting or lying exist apart from that during awake-time for the whole day, yes, then SitLie_30min_bouts_TH = 90 min. 

But keep in mind these variables are designed to be in-line with our perception of a "bout". In other words, when someone says that they  "sat for 90 minutes straight" , it does not necessarily mean that they never stood-up to stretch their legs, adjust camera during a meeting, picked up document from a drawer etc. This means SitLie_30min_bouts_TH could be lower than 90 minutes. Imagine a cashier who sits and stands regularly during their job. According to ActiPASS (if we use defaults which only use bout-breaks, but not thresholds), this person would have perceived SitLie bouts which are very long. And the variable **SitLie_30min_bouts_TH** will be more or less equal to variable **SiltLie** during work-hours.

Sometimes, this derivation may not be what we want. Imagine above cashier alternates between  sit, stand and move. However, even then, the variable **SitLie_30min_bouts_TH** could be almost equal to the variable **SiltLie** during work hours. Does this mean their work hours entirely comprise of only sitting bouts longer than 30 minutes? So, this is a perception question. And sometimes, simple definition we use might not be what we really want (we then need to also use bout-thresholds as described above).

In occupational health, for the behaviour SitLie", we would like to know actual bout lengths. Also, for the behaviour "SitLie" there is no ambiguity about it algorithm wise (unlike for an activities like walking/moving/standing or intensity-classes MVPA). therefore in occupational health studies we normally use variables like **SitLie_T30min** which are strict bouts (not perceived bouts).

**Comment by pasan-h** [comment]
Also see: https://github.com/Ergo-Tools/ActiPASS/discussions/129
